packages/client/src/object/getLinkedObjectOrThrow.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ObjectTypeKeysFrom, OntologyDefinition } from "@osdk/api";
+import { type ClientContext, PalantirApiError } from "@osdk/shared.net";
+import { pageLinkedObjectsOrThrow } from "./pageLinkedObjectsOrThrow.js";
+
+export async function getLinkedObjectOrThrow<
+  O extends OntologyDefinition<any>,
+  T extends ObjectTypeKeysFrom<O>,
+>(
+  client: ClientContext<O>,
+  sourceApiName: string,
+  primaryKey: any,
+  linkTypeApiName: T & string,
+) {
+  const result = await pageLinkedObjectsOrThrow(
+    client,
+    sourceApiName,
+    primaryKey,
+    linkTypeApiName,
+    { pageSize: 1 },
+  );
+
+  if (result.data.length !== 1 || result.nextPageToken != null) {
+    throw new PalantirApiError(
+      `Expected a single result but got ${result.data.length} instead${
+        result.nextPageToken != null ? " with nextPageToken set" : ""
+      }`,
+    );
+  }
```

Comment:
^ is this how we want to handle non-conjure errors related to network requests?

Reply:
I think it doesn't matter in this case. We aren't using any of the other fields. We aren't leveraging retry (nor would i expect doing it again to work). 

Using this vs Error doesnt change anything besides how someone consumes it, right?